src/shared/lib/StoreUtils.ts

```diff
@@ -1622,6 +1624,115 @@ export function makeGetOncoKbCnaAnnotationForOncoprint(
     }
 }
 
+export function getSampleClinicalDataMapByThreshold(
+    clinicalData: ClinicalData[],
+    clinicalAttributeId: string,
+    threshold: number
+) {
+    return _.reduce(
+        clinicalData,
+        (acc, next) => {
+            if (next.clinicalAttributeId === clinicalAttributeId) {
+                const value = getNumericalClinicalDataValue(next);
+                if (value && value >= threshold) {
+                    acc[next.sampleId] = next;
+                }
+            }
+            return acc;
+        },
+        {} as { [key: string]: ClinicalData }
+    );
+}
+
+export function getNumericalClinicalDataValue(
+    clinicalData: ClinicalData
+): number | undefined {
+    if (Number.isNaN(clinicalData.value)) {
```

Comment:
Just checking but this assumes that clinicalData.value is numerical (if in string form).  Number.isNaN("blah") is false so you could get Number("blah") which would be NaN.

Reply:
Correct.